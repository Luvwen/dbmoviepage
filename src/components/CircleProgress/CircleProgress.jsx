import { CircularProgress, CircularProgressLabel } from '@chakra-ui/react'
import React from 'react'

export const CircleProgress = ({ vote, top, left }) => {
  return (
    <>
      <CircularProgress
        position={['relative']}
        value={vote}
        top={top}
        left={left}
        color={vote > 75 ? 'blue.500' : 'yellow.400'}
      >
        <CircularProgressLabel
          color={vote > 75 ? 'green.300' : 'orange.300'}
          bg='blackAlpha.900'
          borderRadius='50%'
          width='75%'
          height='75%'
          lineHeight='10'
          fontWeight='bolder'
        >
          {vote}%
        </CircularProgressLabel>
      </CircularProgress>
    </>
  )
}
