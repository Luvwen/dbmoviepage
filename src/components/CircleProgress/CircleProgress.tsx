import { CircularProgress, CircularProgressLabel } from '@chakra-ui/react';

interface Props {
    vote: number;
    top?: string;
    left?: string;
}

export const CircleProgress: React.FC<Props> = ({ vote, top, left }) => {
    return (
        <>
            <CircularProgress
                as="div"
                color={vote > 75 ? 'blue.500' : 'yellow.400'}
                data-testid="circle div"
                left={left}
                position={['relative']}
                top={top}
                value={vote}
            >
                <CircularProgressLabel
                    bg="blackAlpha.900"
                    borderRadius="50%"
                    color={vote > 75 ? 'green.300' : 'orange.300'}
                    fontWeight="bolder"
                    height="75%"
                    lineHeight="10"
                    width="75%"
                >
                    {vote}%
                </CircularProgressLabel>
            </CircularProgress>
        </>
    );
};
