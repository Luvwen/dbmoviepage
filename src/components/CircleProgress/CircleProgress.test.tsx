import { render, screen } from '@testing-library/react';
import { CircleProgress } from './CircleProgress';

describe('tests in CircleProgress component', () => {
    const vote = 65;

    test('should appear the vote percent', () => {
        render(<CircleProgress vote={vote} />);
        const circleDiv = screen.getByText(`${vote}%`);
        expect(circleDiv.innerHTML).toBe(`${vote}%`);
    });

    test('should match the snapshot', () => {
        const { container } = render(<CircleProgress vote={vote} />);
        expect(container).toMatchSnapshot();
    });
});
