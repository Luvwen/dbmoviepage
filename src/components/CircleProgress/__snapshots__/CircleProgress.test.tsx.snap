// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`tests in CircleProgress component should match the snapshot 1`] = `
<div>
  <div
    aria-valuemax="100"
    aria-valuemin="0"
    aria-valuenow="65"
    class="chakra-progress css-120wkjd"
    data-testid="circle div"
    role="progressbar"
  >
    <svg
      class="css-kn46u7"
      viewBox="0 0 100 100"
    >
      <circle
        class="chakra-progress__track css-1bhxzgy"
        cx="50"
        cy="50"
        r="42"
        stroke-width="10px"
      />
      <circle
        class="chakra-progress__indicator css-u8otmg"
        cx="50"
        cy="50"
        r="42"
        stroke-dasharray="171.6 92.4"
        stroke-dashoffset="66"
        stroke-width="10px"
      />
    </svg>
    <div
      class="css-19ir226"
    >
      65
      %
    </div>
  </div>
</div>
`;
